server/graphql/common/orders.ts

```diff
@@ -106,7 +107,26 @@ export async function addFunds(order: AddFundsInput, remoteUser: User) {
   // Added Funds are not eligible to Platform Tips
   orderData.platformTipEligible = false;
 
-  const orderCreated = await models.Order.create(orderData);
+  // Check transactions import row
+  if (order.transactionsImportRow) {
+    const transactionsImport = await order.transactionsImportRow.getImport();
+    if (!transactionsImport) {
+      throw new NotFound('TransactionsImport not found');
+    } else if (transactionsImport.CollectiveId !== host.id) {
+      throw new ValidationFailed('This import does not belong to the host');
+    } else if (order.transactionsImportRow.isProcessed()) {
+      throw new ValidationFailed('This import row has already been processed');
```

Comment:
Maybe move that condition after line 111, no need to fetch transactionsImport for that.